src/rgw/rgw_sal_rados.h

```diff
@@ -253,7 +253,7 @@ class RGWRadosBucket : public RGWBucket {
 			   bool *is_truncated, RGWUsageIter& usage_iter,
 			   map<rgw_user_bucket, rgw_usage_log_entry>& usage) override;
     virtual std::unique_ptr<RGWBucket> clone() {
-      return std::move(std::unique_ptr<RGWBucket>(new RGWRadosBucket(*this)));
+      return std::unique_ptr<RGWBucket>(new RGWRadosBucket(*this));
```

Comment:
nit, use `std::make_unique<>` instead? not strictly in the scope of this PR, though.

Reply:
@tchaikov 
Let's do that in another PR, since a first try at this gives me lots of complaints about things like:
```
 note: unimplemented pure virtual method 'list' in 'RGWBucket'
    virtual int list(ListParams&, int, ListResults&, optional_yield y) = 0;
```
So that'll be more than just this simple replace.